pkg/nvml/cgo_helpers_static.go

```diff
@@ -73,3 +73,7 @@ func unpackPCharString(str string) (*C.char, *struct{}) {
 	h := (*stringHeader)(unsafe.Pointer(&str))
 	return (*C.char)(h.Data), cgoAllocsUnknown
 }
+
+func VgpuTypeIdToUint32(typeId VgpuTypeId) uint32 {
```

Comment:
I know that there are exceptions, but in general we try to expose NVML-functionality in these bindings and this goes against that a bit.

Does your example not indicate the need for a `String()` function off interface instead of an explicit `*ToUint32` function?

Reply:
I got impression that interface for VgpuTypeId is auto generated based on NVML lib funcs, so I wanted to propose less intrusive option. `String` or `uint32` both works for me. I wanted to show that newer `go-nvml` version is missing some info that was present in older version.

Reply:
I have looked at this a bit. What about doing something like we do for the `Interface` type where we have the function:
```
Extensions() ExtendedInterface
```

We could to the same for `VgpuTypeId`:
```
//go:generate moq -rm -out mock/extendedvgputypeid.go -pkg mock . ExtendedVgpuTypeId:ExtendedVgpuTypeId
type ExtendedVgpuTypeId interface {
       ID() uint32
}
```

I've pushed a commit with a proposal on top. Let me know what you think.

Reply:
Looks good to me! Thanks for looking into this.